---
title: 'Anaylsis of Shanghai Data from NCBI SARS-CoV-2 SRA Bioproject ID 627662'
author: "Mikaela Kuan"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "data/metadata/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview

This is a report on SARS-CoV-2, including some variant analysis [@koyama2020variant].

# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures

```{r snp-call-quality}
vcf_with_metadata %>%
  ggplot(aes(x = as.numeric(qual))) +
    geom_histogram() +
    labs(title = "Frequency of the Quality Scores in the SNPs from Shanghai",
         x = "Quality Score",
         y = "Frequency") +
    theme_few()
    

```
**Figure 1** The frequencies of the quality scores from Shanghai data. Quality scores less than 200 are not of high quality.


# Tables

```{r snp-frequency}
# This is a figure to show the frequency of each SNP
vcf_with_metadata %>%
  select(pos, ref, alt) %>%
  group_by(pos) %>%
  tally() %>%
  filter(n > 5) %>%
  knitr::kable(col.names = c("SNP Position",
                             "Frequency"))
```

**Table 1**: Frequency of each SNP position. The highest frequency was the SNPs in position 28144 and 8782. SNPs with frequencies less than five were removed from the table. 

```{r snp-pos-28144}
pos_1 <- vcf_with_metadata %>%
  select(pos, ref, alt) %>%
  filter(pos == "28144") %>%
  group_by(pos, ref, alt) %>%
  tally()
  
pos_2 <- vcf_with_metadata %>%
  select(pos, ref, alt) %>%
  filter(pos == "8782") %>%
  group_by(pos, ref, alt) %>%
  tally()

pos_3 <- vcf_with_metadata %>%
  select(pos, ref, alt) %>%
  filter(pos == "29742") %>%
  group_by(pos, ref, alt) %>%
  tally()

pos_4 <- vcf_with_metadata %>%
  select(pos, ref, alt) %>%
  filter(pos == "11083") %>%
  group_by(pos, ref, alt) %>%
  tally()

rbind(pos_1, pos_2, pos_3, pos_4) %>%
  knitr::kable(col.names = c("SNP Position",
                             "Reference Nucleotide",
                             "Altered Nucleotide", 
                             "Frequency"))
```

**Table 2** Frequency of the SNPs and its different changes of the top four frequent SNPs. 

# Sources Cited
